### PR TITLE
doc: add container block documentation

### PR DESCRIPTION
**Pull request by bettio**
Add documentation for container block based on http_source doc template.

### CODE REVIEW

File: guides/0111-container.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+This is a special block which allows to offload computation to a
+[Docker container](https://www.docker.com/resources/what-container). 
+
+Messages are sent and received to/from the container via
+[AMQP](https://www.rabbitmq.com/tutorials/amqp-concepts.html).
+
+The block will manage the creation of the Container in a [Kubernetes](https://kubernetes.io/)
+cluster using the
+[Astarte Kubernetes Operator](https://github.com/astarte-platform/astarte-kubernetes-operator).
+
```
Comment (rbino) [inline guides/0111-container.md:14]: Extra newline

File: guides/0111-container.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
+# container
+
+**This block API will likely change in future versions.**
+
+This is a special block which allows to offload computation to a
+[Docker container](https://www.docker.com/resources/what-container). 
+
+Messages are sent and received to/from the container via
+[AMQP](https://www.rabbitmq.com/tutorials/amqp-concepts.html).
```
Comment (rbino) [inline guides/0111-container.md:9]: Add a paragraph mentioning Astarte Flow SDK, something like

```
To simplify the development of a Docker image that will be executed by a container block,
you can use Astarte Flow SDK (currently available for
[Python](https://github.com/astarte-platform/astarte_flow_sdk_python)). The SDK will take care of
receiving and publishing messages to the appropriate RabbitMQ exchanges and queues while the only
part that will be left to be implemented is the data handling for your specific usecase.
```

File: guides/0111-container.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+## `image`
+
+Docker container image.
+
+## `type`
+
+**This block property is going to be removed in favor of implicit configuration.**
+
+According to container type a container is used as an AMQP producer, cosumer or both.
```
Comment (rbino) [inline guides/0111-container.md:29]: "or both" -> "or as a middle processing block"

File: guides/0111-container.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+# Input message
+
+Input messages are consumed only from `consumer` and `producer_consumer` containers.
+
+For those types of containers the input message format depends on their implementation.
```
Comment (rbino) [inline guides/0111-container.md:35]: -> The format of the input message depends on the output format of the last block before it

**Review by rbino — CHANGES_REQUESTED** [review]

File: guides/0111-container.md (hunk 1 above)
Comment (bettio) [reply guides/0111-container.md:14]: this is for a new paragraph

**Review by rbino — APPROVED** [review]

**Pull request merged**